common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -714,6 +714,7 @@ def user_details_force_sync(auth_entry, strategy, details, user=None, *args, **k
         field_mapping.update({
             'fullname': (user.profile, 'name'),
             'country': (user.profile, 'country'),
+            'gender': (user.profile, 'gender'),
```

Comment:
I think this might not work for some cases. Different systems might represent this differently. Open edX only supports `m` for Male, `f` for Female and `o` for Other. If the external system returns these values in a different format then we might have an issue. 

Not sure what the solution here is. I think one way is to allow customising this via some api.

Reply:
@xitij2000, I'd say the middleware built to add an OAuth2 backend to integrate with Open edX should translate the data to the format that Open edX expects. In this case, the Drupal backend does exactly that. So is this really a concern?

Reply:
The concern is about upstreamability. This particular backend does what we expect. Do the other ones in the platform do so as well? What about custom backends installed by others?

Reply:
@xitij2000, since we are just adding a new profile field, I'd say the existing custom backends wouldn't be trying to set the `gender` field via this code anyway. And the ones that try to do so in the future once we upstream this, will have to follow the format expected by that field. And afaik, this field is optional and hence shouldn't break any existing backends. Does this make sense?

Reply:
@lgp171188  I mention this because I just recently implemented an OAuth2 based backend for SSO, and it simply passed the data returned from the site's user info api to the platform. It's probably not the best approach, but if a custom, third-party, or even [core](https://github.com/python-social-auth/social-core/tree/master/social_core/backends) backend for django includes this info and it's in the wrong format, it will cause issues. 

I think one way would be to make this configurable or a per-backend basis, i.e. if a backend reports that is supports the correct format then extract this (and potentially even more data) otherwise not.

Reply:
> I think one way would be to make this configurable or a per-backend basis, i.e. if a backend reports that is supports the correct format then extract this (and potentially even more data) otherwise not.

@xitij2000, if you have any idea on whether we have something like this already in the platform or how this could be done, can you post the details on the task?

> Either way I think we can merge this for now, but have a discovery for the path forward.
:+1: . FYI @kaizoku

Reply:
> @xitij2000, if you have any idea on whether we have something like this already in the platform or how this could be done, can you post the details on the task?

I don't think anything like this is supported ATM.